File: runme/mcgen/local/datan/exasrc/e2gam.c
#include <stdio.h>
#include "datsrc.h"

main(void)
{
    /* Local variables */
    static integer i, ires[12], k, n, nx;
/* identify program to user */
    printf("%s\n", "Program E2GAM demonstrates use of GBINCO");
    printf("%s\n", "and computes and prints ");
    printf("%s\n\n", "top of Pascal""s triangle.");
/* loop over N */
    for (n = 0; n <= 11; ++n) {
/* loop over K */
	for (k = 0; k <= n; ++k) {
/* compute binomial coefficient */
	    ires[k] = (integer) gbinco_(&n, &k);
	}
/* compute format */
	nx = (11 - n) * 3 + 1;
        for (i = 1; i <= nx; ++i)
            printf(" ");
/* write one line of triangle */
        for (i = 1; i <= n+1; ++i)
            printf("%6li", ires[i-1]);
        printf("\n");
    }
    return 0;
} /* main */

File: runme/mcgen/local/datan/exasrc/e2gam.f
      PROGRAM E2GAM
      IMPLICIT DOUBLE PRECISION(A-H,O-Z)
      CHARACTER*10 FORM
      PARAMETER(NMAX=11)
      DIMENSION IRES(NMAX+1)
      DATA FORM /'(**X,15I6)'/
C identify program to user
      WRITE(*,*)' Program E2GAM demonstrates use of GBINCO'
      WRITE(*,*)' and computes and prints '
      WRITE(*,*)' top of Pascal''s triangle'
      WRITE(*,*)' '
C loop over N
      DO 20 N=0,NMAX
C loop over K
        DO 10 K=0,N
C compute binomial coefficient
          IRES(K+1)=INT(GBINCO(N,K))
10      CONTINUE
C compute format
        NX=(NMAX-N)*3+1
        WRITE(FORM(2:3),'(I2)')NX
C write one line of triangle
        WRITE(*,FORM)(IRES(J),J=1,N+1)
20    CONTINUE
      END
